webpack.conf.js

```diff
@@ -1,28 +1,58 @@
-var webpack = require('webpack');
+var CleanWebpackPlugin = require('clean-webpack-plugin'),
+    config = require('./config.json'),
+    path = require('path'),
+    webpack = require('webpack');
 
 module.exports = {
+    watch: false,
     devtool: 'source-map',
+    context: __dirname,
+    entry: {
+        main: './assets/js/app.js',
+    },
+    output: {
+        filename: 'theme-bundle.[name].js',
+        path: path.resolve(__dirname, "assets/dist")
+    },
     bail: true,
     module: {
-        loaders: [
+        rules: [
             {
                 test: /\.js$/,
-                loader: 'babel',
                 include: /(assets\/js|assets\\js|stencil-utils)/,
-                query: {
-                    compact: false,
-                    cacheDirectory: true,
-                    presets: ['es2015-loose'],
+                use: {
+                    loader: 'babel-loader',
+                    options: {
+                        compact: false,
+                        cacheDirectory: true,
+                        presets: [['es2015', {loose: true, modules: false}]],
+                        plugins: [
+                            'dynamic-import-webpack',
+                            'syntax-dynamic-import',
+                            'transform-async-to-generator',
+                            'transform-regenerator'
+                        ]
+                    }
                 }
             }
         ]
     },
     plugins: [
+        new CleanWebpackPlugin(['assets/dist'], {
+            verbose: true,
+            watch: false
+        }),
         new webpack.ProvidePlugin({
             $: 'jquery',
             jQuery: 'jquery',
             'window.jQuery': 'jquery'
+        }),
+        new webpack.optimize.CommonsChunkPlugin({
+            name: 'vendor',
+            minChunks: function (module) {
```

Comment:
https://webpack.js.org/plugins/commons-chunk-plugin/#passing-the-minchunks-property-a-function